### ISSUE TITLE
Add more validator in `Validators`

### ISSUE THREAD

**Issue by kasir-barati** [angular/angular#46427]
### Which @angular/* package(s) are relevant/related to the feature request?

forms

### Description

I am really wondered why we should have `Validators` in the `@angular/forms` while it has less than satisfactory proficiency as I can measure by counterpoising it to something like [class-validator](https://github.com/typestack/class-validator). I am not devaluing `Validators`. I am just saying when we finally have to install [3rd party packages](https://openbase.com/categories/js/best-angular-validation-libraries) due to lake of `Validators` functionality why even it should exists?

I am just asking question. If you ask me I would say I prefer to use Angular built-in functionalities if it gave me what I need. BTW this is my opinion and please tell me if I am wrong.

### Proposed solution

Add more built-in validators. here is a simple list which I would say pretty much every developer in his app needs:

- isBoolean
- isAlphanumberic
- isAlpha
- isUppercase
- isLowercase
- validateIf // as its name implies it will validate if a given condition be true. Mostly that condition emerges from the form
- isNumber
- isInt32
- isInt64
- isFloat
- isDouble
- isString
- isDateString
- isDate

I know some of them seems unreasonable but this is my opinion.

I am inspired by class-validator.

### Alternatives considered

There is no problem with publishing uncomplete API where there are plans to make it better - Kaizen.
But if there is no plan to make it better and other libraries did/do not build their libraries on top of it, IMHO it is better to turn it off.

Note: Even though I said and say this is a #completely_personal_opinion. But I feel righteous indignation :joy: just kidding.

**Comment by dylhunn** [comment]
Hi Kasir, thanks for opening this issue. Our current validators are based on the set of native HTML validators. After some discussion, the team wants to keep it to a small set, and encourage you to pull in or write your own, if you need additional validators.

**Comment by kasir-barati** [comment]
Hmmm. ETC\* principle is good for times that we do not know what we want not when we know what do we exactly need. BTW I am coding some custom validators.

TBH IDK is it good to expand and add moe functionality inside the framework or not. But at the end of the day we all need to deal with forms most of our times. So IMO writing or even copy pasting some custom validators is not a good plan for a long term problem. I prefer to use a core module which needs less maintenance. BTW I am not sure and just to elaborate what do I mean by preferring using core module over my/others custom validator you can see at these different approaches: 
- https://www.npmjs.com/package/angular-form-validator
- https://www.npmjs.com/package/angular-reactive-validation
- https://www.npmjs.com/package/ngx-custom-validators
- https://www.npmjs.com/package/@rxweb/reactive-form-validators

Basically I feel a little unsure now :confused:. Is it good to let people use their own validators or using a robust built-in validator :thinking:.

\* ETC stands for Easy To Change. I feel you meant this.

@dylhunn
